### PR TITLE
Document version macros

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by dalg24** [comment]
(Fixed typo `if (KOKKOS_VERSION_PATCH = 99)` -> `==`)

File: docs/source/API/core/Macros.rst [hunk 1]
```diff
@@ -4,11 +4,71 @@ Macros
...
-| ``KOKKOS_VERSION`` | The Kokkos version; ``KOKKOS_VERSION % 100`` is the patch level, ``KOKKOS_VERSION / 100 % 100`` is the minor version, and ``KOKKOS_VERSION / 10000`` is the major version.   |
-+--------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+.. list-table::
+   :header-rows: 1
+
+   * - Version integer macros
+   * - ``KOKKOS_VERSION``
+   * - ``KOKKOS_VERSION_MAJOR``
+   * - ``KOKKOS_VERSION_MINOR``
+   * - ``KOKKOS_VERSION_PATCH``
```
Comment (masterleinad) [inline docs/source/API/core/Macros.rst:14]: I don't like the looks of the one column tables very much and would either replace them with a list or add a second column that demonstrates their definition for some release.

File: docs/source/API/core/Macros.rst [hunk 2]
```diff
@@ -4,11 +4,78 @@ Macros
...
-| ``KOKKOS_VERSION`` | The Kokkos version; ``KOKKOS_VERSION % 100`` is the patch level, ``KOKKOS_VERSION / 100 % 100`` is the minor version, and ``KOKKOS_VERSION / 10000`` is the major version.   |
-+--------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+.. list-table::
+   :header-rows: 1
+   :align: left
+
+   * - Version integer macros
+     - Example value assuming Kokkos version 4.2.1
+   * - ``KOKKOS_VERSION``
+     - ``4'02'01``
```
Comment (masterleinad) [inline docs/source/API/core/Macros.rst:14]:
```suggestion
     - ``40201``
```

Comment (dalg24) [reply docs/source/API/core/Macros.rst:14]: Why?

Comment (masterleinad) [reply docs/source/API/core/Macros.rst:14]: That's the literal value, isn't it?

Comment (dalg24) [reply docs/source/API/core/Macros.rst:14]: From https://en.cppreference.com/w/cpp/language/integer_literal

> Optional single quotes (`'`) may be inserted between the digits as a separator; they are ignored when determining the value of the literal.

Comment (masterleinad) [reply docs/source/API/core/Macros.rst:14]: I now that but I think it's more confusing for the average user looking at this page.

Comment (dalg24) [reply docs/source/API/core/Macros.rst]: I dropped the separators

**Review by masterleinad — APPROVED** [review]

**Pull request merged**